pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -105,18 +104,29 @@ stdenv.mkDerivation (finalAttrs: {
   ++ lib.optionals openclSupport [
     "-DLLAMA_CLBLAST=ON"
   ]
-  ++ lib.optionals openblasSupport [
+  ++ lib.optionals blasSupport [
     "-DLLAMA_BLAS=ON"
     "-DLLAMA_BLAS_VENDOR=OpenBLAS"
+  ]
+  ++ lib.optionals (!static) [
+    (lib.cmakeBool "BUILD_SHARED_LIBS" true)
   ];
 
   installPhase = ''
     runHook preInstall
 
     mkdir -p $out/bin
+    ${lib.optionalString (!static) ''
+      mkdir $out/lib
+      cp libggml_shared.so $out/lib
+      cp libllama.so $out/lib
```

Comment:
There are cmake install rules for both, just delete the custom install phase?

Reply:
Could you specify a bit which rules you are talking about, I'm not super familiar with the c ++ build system.
I've seen something that could correspond in the makefile
https://github.com/ggerganov/llama.cpp/blob/master/Makefile#L614
But I'm not sure how to make nix install this in the end.
I've tried adding "installTargets" but it didn't work.
(originally I added those since the ollama tool is going in the direction of requiring the .so files).
Happy to make a PR for a change here if you have more details.

Reply:
https://github.com/ggerganov/llama.cpp/blob/099afc6274c859ca67146e725839f2d97a5ef313/CMakeLists.txt#L1051-L1056
https://github.com/ggerganov/llama.cpp/blob/099afc6274c859ca67146e725839f2d97a5ef313/CMakeLists.txt#L1124

Reply:
then what else needs to be changed on the nix side to make those appear in the lib directory ?

just reading the cmakefile it seems it installs inside a LIBRARY directory, is there another env var that needs to be set ? or maybe TARGETS ?
I've tried by just removing the step and they don't appear.

Reply:
The setup hooks will set up reasonable prefixes and install paths, they only rarely need modification

Reply:
> I've tried by just removing the step and they don't appear.

H'm, I'd need to look into this. But do check out the flake's `package.nix`. It's got much clearer with Philip's PR upstream